### PR TITLE
[FE] [ADD] Enable dark mode for both Web

### PR DESCRIPTION
**Pull request by RenanDbs**
### 🛠 Changes being made
Posibility to enable dark mode for both on the web in the MyAccount pages
### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/restoWeb/src/pages/AddDishPage/AddDishPage.tsx [hunk 1]
```diff
@@ -1,13 +1,34 @@
-import React from "react";
-
+import React, {useEffect, useState} from "react";
 import DishForm from "@src/components/forms/DishForm/DishForm";
 import Layout from 'shared/components/Layout/Layout';
 import styles from "@src/pages/AddDishPage/AddDishPage.module.scss";
+import { enable, disable, setFetchMethod} from "darkreader";
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/AddDishPage/AddDishPage.tsx:5]: darkreader needs to be installed via package.json

File: packages/restoWeb/src/pages/AddDishPage/AddDishPage.tsx [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
 import {useTranslation} from "react-i18next";
 
 const AddDishPage = () => {
   const {t} = useTranslation();
+  
+  useEffect(() => {
+    checkDarkMode();
+  }, []);
 
+  const checkDarkMode = () => {
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/AddDishPage/AddDishPage.tsx:15]: maybe we can move this method to the shared folder to reduce the amount of duplicate code?

File: packages/visitorWeb/src/pages/AboutUsPage/AboutUsPage.tsx [hunk 3]
```diff
@@ -44,6 +45,59 @@ const AboutUsPage = () => {
     },
   ];
 
+  const description = `Guardos is your trusted companion in the world of dining,
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/AboutUsPage/AboutUsPage.tsx:48]: description, foundingstory and teamsdesciption should not be added, I think this is a merge error from the i18n ticket.

File: packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 4]
```diff
@@ -475,6 +508,8 @@ const MyAccountPage = () => {
           {t('pages.MyAccountPage.delete-account')}
         </button>
         <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: '20px' }}>
+          <Button onClick={toggleDarkMode}>{localStorage.getItem('darkMode') === 'true' ? 'Light Mode' : 'Dark Mode'}</Button>
+          <Typography variant="body1">You need a new feature? </Typography>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx:512]: also merge error, the typography with hardcoded "You need a new feature?" should not be added.

File: packages/restoWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 5]
```diff
@@ -431,6 +467,8 @@ const MyAccountPage = () => {
           {t('pages.MyAccountPage.delete-account')}
         </button>
         <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: '20px' }}>
+          <Button onClick={toggleDarkMode}>{localStorage.getItem('darkMode') === 'true' ? 'Light Mode' : 'Dark Mode'}</Button>
+          <Typography variant="body1">You need a new feature? </Typography>
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/MyAccountPage/MyAccountPage.tsx:471]: merge error, the typography with hardcoded "You need a new feature?" should not be added.

File: packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 6]
```diff
@@ -475,6 +508,8 @@ const MyAccountPage = () => {
           {t('pages.MyAccountPage.delete-account')}
         </button>
         <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: '20px' }}>
+          <Button onClick={toggleDarkMode}>{localStorage.getItem('darkMode') === 'true' ? 'Light Mode' : 'Dark Mode'}</Button>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx:511]: I think a toggle button would be more fitting here ("Enable dark mode")

File: packages/restoWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 7]
```diff
@@ -431,6 +467,8 @@ const MyAccountPage = () => {
           {t('pages.MyAccountPage.delete-account')}
         </button>
         <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: '20px' }}>
+          <Button onClick={toggleDarkMode}>{localStorage.getItem('darkMode') === 'true' ? 'Light Mode' : 'Dark Mode'}</Button>
```
Comment (JosefineM2000) [inline packages/restoWeb/src/pages/MyAccountPage/MyAccountPage.tsx:470]: I think a toggle button would be more fitting here ("Enable dark mode")

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]
Please check for several browser that this works (tested in firefox did not work)

**Review by JosefineM2000 — APPROVED** [review]
